File: Maps/CheckCloseString/close.cpp
#include <string>
#include <unordered_map>
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    bool closeStrings(string word1, string word2) {
        int n1 = word1.length();
        int n2 = word2.length();
        if(n1 != n2)return false;
        unordered_map<char,int>mp1,mp2;
        for(int i=0;i<n1;i++){
            mp1[word1[i]]++;
            mp2[word2[i]]++;
        }

        if (mp1.size() != mp2.size())
            return false;

        for (auto &p : mp1) {
            if (mp2.find(p.first) == mp2.end())
                return false;
        }
        
        vector<int> f1, f2;

        for (auto p : mp1)
            f1.push_back(p.second);

        for (auto p : mp2)
            f2.push_back(p.second);

        sort(f1.begin(), f1.end());
        sort(f2.begin(), f2.end());

        return f1 == f2;
    }
};

File: Maps/CheckCloseString/close.py
class Solution:
    def closeStrings(self, word1: str, word2: str) -> bool:
        n1 = len(word1)
        n2 = len(word2)
        if n1 != n2: return False
        mp1,mp2 = {},{}
        for i in range(n1):
            mp1[word1[i]] = mp1.get(word1[i],0) + 1
            mp2[word2[i]] = mp2.get(word2[i],0) + 1

        if len(mp1) != len(mp2):
            return False

        for key,value in mp1.items():
            if key not in mp2:
                return False
        
        f1,f2 = [],[]

        for key,value in mp1.items():
            f1.append(value)

        for key,value in mp2.items():
            f2.append(value)

        f1.sort()
        f2.sort()

        return f1 == f2